jekyll/_posts/2018-06-09-discover-godot-nim.md

```diff
@@ -0,0 +1,93 @@
+## Discovering Godot and Nim
+
+In my ongoing quest to get better at programming and game development, I stumbled upon [Nim](https://nim-lang.org/) and it was love at first "Hello, World!". Meanwhile a little bird whispered news to me that a cool open-source game engine, called [Godot](https://godotengine.org/), is reaching 3.0 status with a lot of new features, among others a way to include scripting in languages that aren't builtin. Eventually Nim support landed for Godot and I thought it was the perfect time to get started. Here follows my experiences with Nim and learning Godot.
+
+### Enter Nim
+
+```nim
+  echo("Hello, World!")
+```
+
+As a beginner and hobbyist programmer, Nim was easy to learn but hard to understand. The [main tutorial](https://nim-lang.org/0.18.0/tut1.html) and [book](https://www.manning.com/books/nim-in-action) are excellent starting resources and I could get productive without knowing the inner workings of Nim.
+My biggest hurdle was thinking in the procedural programming realm as most beginner resources I read before were OOP-orientated. Once that and Nim's straightforward type system clicked I found myself doing coding challenges and starting up small projects(like an interpreter), instead of browsing StackOverflow to copy-and-paste my solution.
+Programming in Nim is fun for me, because of the clear syntax, fast compile times and helpful community. I haven't even gotten to the juicy parts of Nim: metaprogramming!
+
+```nim
+# This is my solution for the first Rosalind problem
+# also one of the first Nim snippets I wrote
+import tables
+
+const input = "AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGAT"
+var symbolCount = initCountTable[char]()
+
+for c in input:
+    symbolCount.inc(c, 1)
+
+for k, v in symbolCount:
+    stdout.write v, " "
+```
+
+#####
```

Comment:
Is this valid Markdown? Horizontal lines should typically be created using `___` (three underscores).

Reply:
I have no idea my markdown-fu is very weak, but in the markdown preview looked OK to me.